### PR TITLE
Fixed windows serial number endianness so processes table shows correct serial number

### PR DESCRIPTION
**Pull request by dthompson12315**
To submit a PR please make sure to follow the next steps:
vmware_2018-10-28_21-23-55

- [x] Read the `CONTRIBUTING.md` guide on the root of the repo.
- [x] Ensure your PR contains a single logical change.
- [x] Ensure your PR contains tests for the changes you're submitting.
- [x] Describe your changes with as much detail as you can.
- [x] Link any issues this PR is related to.
- [x] Remove the text above.

### CODE REVIEW

File: osquery/tables/system/linux/rpm_packages.cpp [hunk 1]
```diff
@@ -150,6 +150,7 @@ QueryData genRpmPackages(QueryContext& context) {
     r["size"] = getRpmAttribute(header, RPMTAG_SIZE, td);
     r["sha1"] = getRpmAttribute(header, RPMTAG_SHA1HEADER, td);
     r["arch"] = getRpmAttribute(header, RPMTAG_ARCH, td);
+    r["epoch"] = getRpmAttribute(header, RPM_EPOCH, td);
```
Comment (muffins) [inline osquery/tables/system/linux/rpm_packages.cpp]: Looks like you've caught some other code changes up in this? Perhaps try rebasing first before we move forward to ensure you're only modifying the certificates table.

File: osquery/tables/system/windows/certificates.cpp [hunk 2]
```diff
@@ -203,7 +203,17 @@ void enumerateCertStore(const HCERTSTORE& certStore,
                     certContext->pCertInfo->SerialNumber.pbData +
                         certContext->pCertInfo->SerialNumber.cbData),
         back_inserter(serial));
-    r["serial"] = serial;
+//serial number was in wrong endian, so reversed the string and then swapped
```
Comment (muffins) [inline osquery/tables/system/windows/certificates.cpp]: You'll wanna space these comments, but I believe you might want to just `clang-format` your commit and you should be good. You can do so with:
`clang-format -i osquery/tables/system/windows/certificates.cpp`

File: osquery/tables/system/windows/certificates.cpp [hunk 3]
```diff
@@ -203,7 +203,17 @@ void enumerateCertStore(const HCERTSTORE& certStore,
                     certContext->pCertInfo->SerialNumber.pbData +
                         certContext->pCertInfo->SerialNumber.cbData),
         back_inserter(serial));
-    r["serial"] = serial;
+//serial number was in wrong endian, so reversed the string and then swapped 
+//every other number to make each byte in the correct order
+    std::reverse(serial.begin(), serial.end());
```
Comment (directionless) [inline osquery/tables/system/windows/certificates.cpp]: Huh. I had no idea it came out wrong endian. Oddly, this is the second time today I've seen something like this in the windows depths. Anyhow... the internet suggests this code, which feels a little cleaner to me:
```
BYTE *pbName=pCertContext->pCertInfo->SerialNumber.pbData;
CString theString ;
for (int i = pCertContext->pCertInfo->SerialNumber.cbData - 1; i >= 0 ; i--)
{
  CString hex ;
  hex.Format("%02x", pbName[i]) ;
  theString += hex ;
}
```

**Comment by mike-myers-tob** [comment]
The Windows certificates table serial ID reading was fixed in PR #5742 but I guess nobody was aware that the fix was also in this PR at the time.

I'm going to close this PR, but encourage @dthompson12315 to open a new PR for the changes to the Linux RPM package code, which are unrelated as @muffins pointed out.

**Pull request closed without merging**